paritytech/ink
# Make the default accounts in integration test and e2e the same addresses

## Summary
Closes #1953 
- [n] y/n | Does it introduce breaking changes?
- [n] y/n | Is it dependant on the specific version of `cargo-contract` or `pallet-contracts`?

## Description

### Function
`default_accounts()`

### Initial Status
Implementation Difference

### Issue Description
Default accounts addresses do not match across integration and e2e tests.

### Issue Documentation & Test Case
https://github.com/CoinFabrik/on-ink-integration-tests/tree/main/test-cases/default-accounts

### Current Status
Implementation Difference Corrected. Test Cases in Target Repo Passed. Pull Request Performed to Corresponding Repository.

### Implementation Notes
Now the integration tests mimic the account setup in e2e tests. We changed the name of the accounts "Django" to "Dave" and "Frank" to "Ferdie". On the other hand, there were two accounts in e2e that did not exist in integration tests, accounts “one” and “two”. We added these accounts to integration tests.<br><br>Moreover, since e2e tests were drawing these accounts from the library `sp_keyring::sr25519::Keyring`, we made integration tests depend on the same library in order to account for future changes in this lib.

### Updated Documentation

We updated the original documentation of this issue in [our repository](https://github.com/CoinFabrik/on-ink-integration-tests/tree/main), adding the section [Update on Correcting this Issue](https://github.com/CoinFabrik/on-ink-integration-tests/blob/main/test-cases/default-accounts/README.md#update-on-correcting-this-issue) and informing of the correction.<br><br>In the target repository, we updated the inline documentation. The associated documentation should be updated automatically in these pages: [ink_env test default_accounts](https://paritytech.github.io/ink/ink_env/test/fn.default_accounts.html), [ink_env test DefaultAccounts](https://paritytech.github.io/ink/ink_env/test/struct.DefaultAccounts.html), [ink_e2e AccountKeyring](https://paritytech.github.io/ink/ink_e2e/enum.AccountKeyring.html).

### Testing Guide
In the directory `integration-tests/default_accounts` of the [target directory](https://github.com/paritytech/ink), we include in our pull request a test case showing that the observed implementation difference has been corrected. Note that this test is different from the [original test case in our repo](https://github.com/CoinFabrik/on-ink-integration-tests/tree/main/test-cases/default-accounts), which showed the implementation difference.<br><br>In this directory `integration-tests/default_accounts`, run the following command to run both integration and e2e tests:<br> `cargo test -–features e2e-tests`<br><br>These tests are run in the same function in order to compare the results of both environments.

## Checklist before requesting a review
